python/taichi/ui/window.py

```diff
@@ -126,6 +131,19 @@ def write_image(self, filename):
         """
         return self.window.write_image(filename)
 
+    def copy_depth_buffer_to_ndarray(self, depth: Ndarray):
+        """fetch the depth information of current scene to ti.Ndarray.
+           (support copy from vulkan to cuda/cpu which is a faster version)
+        Args:
+            depth (Ndarray): [window_width, window_height] carries depth information.
+        """
+        if not (isinstance(depth, Ndarray) and len(depth.shape) == 2):
```

Comment:
I think we should check the shape of depth ndarrary here.

Reply:
There is no function method to get resolution of window on python scope, so I judge this on the cpp part (window.cpp).